api/src/pcapi/core/search/backends/algolia.py

```diff
@@ -150,7 +150,24 @@ def __init__(self) -> None:
         self.algolia_venues_client = client.init_index(settings.ALGOLIA_VENUES_INDEX_NAME)
         self.redis_client = current_app.redis_client
 
+    def _is_offers_index_full(self) -> bool:
+        if settings.ALGOLIA_OFFERS_INDEX_MAX_SIZE < 0:
+            return False
+
+        currently_indexed_offers = self.redis_client.hlen(REDIS_HASHMAP_INDEXED_OFFERS_NAME)
+        offers_in_indexing_queue = self.redis_client.scard(REDIS_OFFER_IDS_NAME)
+        limit = settings.ALGOLIA_OFFERS_INDEX_MAX_SIZE
+
+        return currently_indexed_offers + offers_in_indexing_queue >= limit
+
     def enqueue_offer_ids(self, offer_ids: Iterable[int]) -> None:
+        if self._is_offers_index_full():
+            logger.error(
+                f"Maximum index size (%s) exceeded." % settings.ALGOLIA_OFFERS_INDEX_MAX_SIZE,
```

Comment:
f-string ou "%s", il faut choisir. :) Proposition : 

```suggestion
                f"Maximum index size ({settings.ALGOLIA_OFFERS_INDEX_MAX_SIZE}) exceeded.",
```

Reply:
```suggestion
                "Maximum index size (%s) exceeded.", settings.ALGOLIA_OFFERS_INDEX_MAX_SIZE,
```
pylint demande : `Use lazy % formatting in logging functions`

Reply:
Oui, ça permet à Sentry de regrouper les logs, quelle que soit la valeur de `ALGOLIA_OFFERS_INDEX_MAX_SIZE`. C'est peut-être bien, ou pas, je ne sais pas... :)

Reply:
J'ai pas compris, c'est la f-string ou le lazy-loading qui permet de regrouper les logs ?

Reply:
Regrouper les logs est possible quand les paramètres du message sont passés en argument de la fonction de log : 

```python
logger.error("Erreur sur l'offre %d", offer.id)
```

Dans ce cas, Sentry groupe les logs à partir du message `"Erreur sur l'offre %d"`, qui est toujours le même.

Dans les 2 exemples ci-dessous, par contre, Sentry verra des messages distincts, sans paramètre, et ne regroupera pas : 

```python

logger.error("Erreur sur l'offre %d" % offer.id)
logger.error(f"Erreur sur l'offre {offer.id}")
```

Reply:
Ok, merci pour l'explication !